app/frontend/src/features/profile/useUpdateHostingPreferences.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { act } from "react-test-renderer";
+import { service } from "../../service";
+import { addDefaultUser } from "../../test/utils";
+import useAuthStore from "../auth/useAuthStore";
+import useCurrentUser from "../userQueries/useCurrentUser";
+import useUpdateHostingPreferences from "./useUpdateHostingPreferences";
+import wrapper from "../../test/hookWrapper";
+
+const getUserMock = service.user.getUser as jest.Mock;
+const updateHostingPreferenceMock = service.user
+  .updateHostingPreference as jest.Mock;
+
+beforeAll(() => {
+  // Mock out console.error so the test output is less noisy when
+  // an error is intentionally thrown for negative tests
+  jest.spyOn(console, "error").mockReturnValue(undefined);
+});
+
+describe("useUpdateHostingPreference hook", () => {
+  const newHostingPreferenceData = {
+    multipleGroups: false,
+    acceptsKids: false,
+    acceptsPets: false,
+    lastMinute: true,
+    wheelchairAccessible: true,
+    maxGuests: null,
+    area: "",
+    houseRules: "",
+    smokingAllowed: 1,
+    sleepingArrangement: "",
+  };
+
+  it("updates the store with the latest user hosting preference", async () => {
+    addDefaultUser();
+    const newUserPref = Object.entries(newHostingPreferenceData).reduce(
+      (acc: Record<string, unknown>, [key, value]) => {
+        if (key !== "smokingAllowed") {
+          acc[key] = { value };
+        } else {
+          acc[key] = value;
+        }
+        return acc;
+      },
+      {}
+    );
+    updateHostingPreferenceMock.mockResolvedValue(new Empty());
+    getUserMock.mockImplementation(() => {
+      if (!updateHostingPreferenceMock.mock.calls.length) {
+        return defaultUser;
+      } else {
+        return {
+          ...defaultUser,
+          ...newUserPref,
+        };
+      }
+    });
+    const { result, waitFor } = renderHook(
+      () => ({
+        store: useAuthStore(),
+        mutate: useUpdateHostingPreferences(),
+        currentUser: useCurrentUser(),
```

Comment:
How come we can't render `useUpdateHostingPreferences` on its own like the other tests? Are the other two tests just populating the React Query cache?

Reply:
Do you mean the other two hooks? The reasons is because I use `currentUser` in the assertions below. I can get rid of `store` though.